lib/java/gradle.properties

```diff
@@ -0,0 +1,35 @@
+# This file is shared currently between this Gradle build and the
+# Ant builds for fd303 and JavaScript. Keep the dotted notation for
+# the properties to minimize the changes in the dependencies.
+thrift.version=1.0.0
```

Comment:
There's a document floating around on the thrift web site about releasing Thrift that points folks at what files to update after cutting a release branch; this should be added.  @Jens-G since you went through this recently can you point that out?

Reply:
I can have the CMake or autoconf wrapper system provide the version so this becomes a developer placeholder default. I am surprised this project does manual publishing of the Java artifacts. I can see how much the cross language support has increased the build system complexity but there is room for improvement.

Are all the language bits released by a manual build and process?

Reply:
There is a release process.

https://thrift.apache.org/docs/committers/HowToRelease

Reply:
In that case all we need to provide is the version parameters from Autoconf's own version variables into the execution of all the Java generating tasks. They use this file as their default version to bootstrap for development. This can stay forever at 1.0.0 to facilitate building without make executions. I already wired this project to obey the parsed version for the CMake builds. I can do the same for the Autoconf builds. I do need to find out how the version is expressed in Autoconf makefiles. I saw a PARSED_VERSION variable somewhere, perhaps that is it?

The "make dist" process would then generate all artifacts including the ones for Maven publication and add them to the tar ball. The Maven publication process is intended to be driven by Ant (currently broken) or Gradle and it should correctly post the artifacts as signed artifacts to the Apache Maven staging repository. I believe this could possibly be an automatic process generating "nightlies" for Java and other languages that have a distribution repository available. Nexus has a release promotion process that could then be invoked on any particular "nightly" build that is blessed for release. That would finalize the process that makes the artifacts available at the distribution repositories such as Maven Central.